junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TimeoutExtension.java

```diff
@@ -165,6 +170,26 @@ private ScheduledExecutorService getExecutor(ExtensionContext extensionContext)
 		return extensionContext.getRoot().getStore(NAMESPACE).getOrComputeIfAbsent(ExecutorResource.class).get();
 	}
 
+	private boolean isDebuggingOrDisabled(ExtensionContext extensionContext) {
+		if (extensionContext.getConfigurationParameter(TIMEOUT_MODE_PROPERTY_NAME).isPresent()) {
+			if (extensionContext.getConfigurationParameter(TIMEOUT_MODE_PROPERTY_NAME).get().equals(DISABLED)) {
+				return true;
+			}
+			else if (extensionContext.getConfigurationParameter(TIMEOUT_MODE_PROPERTY_NAME).get().equals(
+				DISABLED_ON_DEBUG)) {
+				for (final String argument : ManagementFactory.getRuntimeMXBean().getInputArguments()) {
```

Comment:
We had issues using `java.lang.management` when this was introduced in junit4:
https://github.com/junit-team/junit4/pull/1187

I think we should wrap `ManagementFactory.getRuntimeMXBean().getInputArguments()` in a try-catch and catch `NoClassDefFoundError` (or whatever is thrown when the class is not present) to make sure this doesn't blow up on Android and other platforms. We should add a test with a custom `ClassLoader` and an integration test that excludes the module from the module path.

Reply:
I'll see what I can do for creating a new test.

Reply:
So I've been giving this a crack over the last couple days and I'm not seeing much progress. Modules and loading them is a little outside my breadth of knowledge. I do want to try some more but do you have some thoughts on where I should start?

Reply:
Here's the source of the integration test that checked the behavior for when module `java.scripting` is missing:

https://github.com/junit-team/junit5/blob/releases/5.4.x/platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/MultiReleaseJarTests.java#L71

Compare the `module-info.java` files in the underlying Maven-based projects at: https://github.com/junit-team/junit5/tree/releases/5.4.x/platform-tooling-support-tests/projects/multi-release-jar -- the one in `default` contains a `requires java.scripting;` directive, the one in `no-scripting` does not.

If you any help, please let me know.

Reply:
That was a good starting point. I can see how the code works up to a certain point. I see Maven's being run and calls all the tests inside each JupiterIntegrationTests class. I don't see how javaScriptingModuleIsAvailable throws any errors but it looks like it actually is.

Reply:
> I don't see how javaScriptingModuleIsAvailable throws any errors but it looks like it actually is.

As the module declared in project `no-scripting` doesn't read `java.scripting` that very module is **not** added to the set of visible modules at run time. As our (deprecated) code only declared a `static` (read: optional) read edge to `java.scripting` we had to ensure that we only use that module, if it is available at run time. That check, which needs to go in this PR, could look similar to this outline:

https://github.com/junit-team/junit5/blob/75907415611e1a8a9bc3e9476231a0d77bc7a23a/junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ScriptExecutionCondition.java#L131-L160

Reply:
Ahh there we go. OK I think I should be able to finish this up tomorrow now. Thanks for the quick responses and they were very helpful as well.

Reply:
I am stuck. I didn't want to pollute this PR so I have created another branch with the integration test [here](https://github.com/chrischild/junit5/tree/issues/integration-test). Specifically [Integration Test](https://github.com/junit-team/junit5/commit/17e982bc6841daef3d76d8cc9fbaef567e190e23) JupiterIntegrationTests and [Customer classloader](https://github.com/junit-team/junit5/commit/7f844fba1b7c9e560952aeaa0f31b8d56dc84c1d) MangementCondition.

Reply:
Sorry I don't think this is something I can figure out on my own. I went so far as to checkout the 5.4.3 code and ran the integration tests. I also changed the class that the ScriptExecutionCondition was trying to load to java.lang.management.ManagementFactory and the class stops throwing the exception that is expected when the java.management module is not in the module-info.java file. It continues on and fails later on when it tries to create an instance of the Evaluator which it should since the java.scripting module isn't loaded. 

Just to be sure I also left the class to be loaded as the ManagementFactory and added the java.scripting package to the no-scripting module-info to see if it would successfully build and it did, so I'm not quite sure where to go from there.

Reply:
Also the CI test is failing sure to a file locking error it looks like.

Reply:
> Also the CI test is failing due to a file locking error it looks like.

I'm fixing this issue at the moment and will push it to `master` soon.

> [...] so I'm not quite sure where to go from there.

I'll have a look at it next week or so. Perhaps, it's the _wrong_ way to test here anyway. As Android doesn't use the module system (does it?) yet, it'll be enough to guard the production code and test the same way as it is done in JUnit 4.

Reply:
Resolved via https://github.com/junit-team/junit5/commit/ce1892cb02b3917d061c414f0b49f08cb6017a52